src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -455,33 +455,71 @@ private int constructMergedTable(Table table, List<FeedToMerge> feedsToMerge,
+                                    String stopCode = stopsReader.get(stopCodeIndex);
+                                    boolean stopCodeIsMissing = "".equals(stopCode);
+                                    if (isRegularStop && stopCodeIsMissing) stopsMissingStopCodeCount++;
+                                }
+                                LOG.info("total stops: {}", stopsCount);
+                                LOG.info("stops missing stop_code: {}", stopsMissingStopCodeCount);
+                                if (stopsMissingStopCodeCount == stopsCount) {
+                                    // If all stops are missing stop_code, we simply default to merging on stop_id.
                                     LOG.warn(
                                         "stop_code is not present in file {}/{}. Reverting to stop_id",
-                                        feedIndex, feedsToMerge.size());
+                                        feedIndex + 1, feedsToMerge.size());
                                     // If the key value for stop_code is not present, revert to stop_id.
                                     keyField = table.getKeyFieldName();
                                     keyFieldIndex = table.getKeyFieldIndex(fieldsFoundInZip);
                                     keyValue = csvReader.get(keyFieldIndex);
-                                    stopCodeMissingFromFirstTable = true;
-                                }
-                            } else {
-                                // Check whether stop_code exists for the subsequent files.
-                                String firstStopCodeValue = csvReader.get(getFieldIndex(fieldsFoundInZip, "stop_code"));
-                                if (stopCodeMissingFromFirstTable && !"".equals(firstStopCodeValue)) {
-                                    // If stop_code was missing from the first file and exists for
-                                    // the second, we consider that a failing error.
+                                    if (feedIndex == 0) {
+                                        stopCodeMissingFromFirstFeed = true;
+                                    }
+                                    if (feedIndex == 1 && !stopCodeMissingFromFirstFeed) {
+                                        mergeFeedsResult.failed = true;
+                                        mergeFeedsResult.errorCount++;
+                                        mergeFeedsResult.failureReasons.add(
+                                            String.format(
+                                                "If stop_code is provided for some stops (for those with location_type = " +
+                                                    "empty or 0), all stops must have stop_code values. The merge process " +
+                                                    "found %d stops that were incorrectly missing stop_code values.",
```

Comment:
The comment might be improved by expressing the number of bad stops out of total stops. For extra extra credit, also omit stations somehow.